cmd/kops/util/factory.go

```diff
@@ -272,6 +277,31 @@ func (f *Factory) buildRESTConfig(ctx context.Context, cluster *kops.Cluster, op
 		options.Admin = kubeconfig.DefaultKubecfgAdminLifetime
 	}
 
+	if options.UseKubeconfig {
+		// Get the kubeconfig from the context
+		klog.Infof("--use-kubeconfig is set; loading connectivity information from kubeconfig (instead of generating it)")
+
+		clusterName := cluster.ObjectMeta.Name
+
+		clientGetter := genericclioptions.NewConfigFlags(true)
```

Comment:
This is restoring the prior logic, which was in 324117cc52943f4c4f6d5a98705aa3e17992edf7